bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/FloureonThermostatHandler.java

```diff
@@ -0,0 +1,252 @@
+import com.github.mob41.blapi.dev.hysen.AdvancedStatusInfo;
+import com.github.mob41.blapi.dev.hysen.BaseStatusInfo;
+import com.github.mob41.blapi.dev.hysen.SensorControl;
+import com.github.mob41.blapi.mac.Mac;
+import com.github.mob41.blapi.pkt.cmd.hysen.SetTimeCommand;
+
+/**
+ * The {@link FloureonThermostatHandler} is responsible for handling thermostats labeled as Floureon Thermostat.
+ *
+ * @author Florian Mueller - Initial contribution
+ */
+@NonNullByDefault
+public class FloureonThermostatHandler extends BroadlinkThermostatHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(FloureonThermostatHandler.class);
+    private @Nullable FloureonDevice floureonDevice;
+
+    /**
+     * Creates a new instance of this class for the {@link FloureonThermostatHandler}.
+     *
+     * @param thing the thing that should be handled, not null
+     */
+    public FloureonThermostatHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * Initializes a new instance of a {@link FloureonThermostatHandler}.
+     */
+    @Override
+    public void initialize() {
+        super.initialize();
+        try {
+            blDevice = new FloureonDevice(host, new Mac(mac));
+            this.floureonDevice = (FloureonDevice) blDevice;
+        } catch (IOException e) {
+            logger.error("Could not find broadlinkthermostat device at Host {} with MAC {} ", host, mac, e);
+            updateStatus(ThingStatus.OFFLINE);
+        }
+        authenticate();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Command: {}", command.toFullString());
+
+        if (command == RefreshType.REFRESH) {
+            refreshData();
+            return;
+        }
+
+        switch (channelUID.getIdWithoutGroup()) {
+            case SETPOINT:
+                handleSetpointCommand(channelUID, command);
+                break;
+            case POWER:
+                handlePowerCommand(channelUID, command);
+                break;
+            case MODE:
+                handleModeCommand(channelUID, command);
+                break;
+            case SENSOR:
+                handleSensorCommand(channelUID, command);
+                break;
+            case REMOTE_LOCK:
+                handleRemoteLockCommand(channelUID, command);
+                break;
+            case TIME:
+                handleSetTimeCommand(channelUID, command);
+                break;
+            default:
+                logger.warn("Channel {} does not support command {}", channelUID, command);
+        }
+    }
+
+    private void handlePowerCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof OnOffType) {
+            try {
+                floureonDevice.setPower(command == OnOffType.ON);
+            } catch (Exception e) {
+                logger.error("Error while setting power of {} to {}", thing.getUID(), command, e);
```

Comment:
Bindings should only log to error if something severe happened, like the detection of a bug in your code. This could be warn. See this link for a description of the log levels: https://www.openhab.org/docs/developer/guidelines.html#f-logging

Also, it seems a bit unreasonable to log the stack trace when the network fails.

Please check all.

Reply:
Changed to warn and only log message.